### ISSUE TITLE
Job analysis template leads to creating jobs repeatelly.

### ISSUE THREAD

**Issue by TommyLike** [argoproj/argo-rollouts#790]
## Summary 

I created a simple job analysis template for http smoke test, but the template belows will leads to argo rollout keep creating the job repeatelly, I am wondering if I misuse the template or this is a potential bug.
```
apiVersion: argoproj.io/v1alpha1
kind: AnalysisTemplate
metadata:
  name: endpoint-reachable
spec:
  metrics:
  - name: endpoint-reachable
    provider:
      job:
        spec:
          backoffLimit: 1
          template:
            spec:
              restartPolicy: Never
              containers:
                - name: api-reachable
                  image: curlimages/curl:7.73.0
                  command:
                    - /bin/sh
                    - -c
                    - |
                      curl -I https://www.baidu.com
                      status_code=$(curl --write-out "%{http_code}\n" --silent --output /dev/null "https://www.baidu.com");
                      [ $status_code -eq 200 ]
```
the command above will success for sure. and the related code lies here:
https://github.com/argoproj/argo-rollouts/blob/master/utils/analysis/helpers.go#L213

the rollout definition:
```
spec:
  replicas: 3
  revisionHistoryLimit: 1
  selector:
    matchLabels:
      app: bluegreen-demo
  strategy:
    blueGreen:
      activeService: bluegreen-demo
      autoPromotionEnabled: false
      prePromotionAnalysis:
        templates:
        - templateName: endpoint-reachable
      previewService: bluegreen-demo-preview
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: bluegreen-demo
    spec:
      containers:
      - image: argoproj/rollouts-demo:blue
        imagePullPolicy: Always
        name: bluegreen-demo
        ports:
        - containerPort: 8080
          name: http
          protocol: TCP
        resources:
          requests:
            cpu: 5m
            memory: 32Mi
```
## Diagnostics

What version of Argo Rollouts are you running?

 **argoproj/argo-rollouts:v0.9.1**

```
# Paste the logs from the rollout controller

# Logs for the entire controller:
kubectl logs -n argo-rollouts deployment/argo-rollouts

# Logs for a specific rollout:
kubectl logs -n argo-rollouts deployment/argo-rollouts | grep rollout=<ROLLOUTNAME>
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

**Comment by mclarke47** [comment]
Can't be sure without seeing the Rollout but it's likely you're using https://argoproj.github.io/argo-rollouts/features/analysis/#background-analysis, which is intended

**Comment by TommyLike** [comment]
@mclarke47 it's `premotionAnalysis`, I can update the issue.

**Comment by jessesuen** [comment]
Can you try with v0.9.2? This had a bunch of fixes with blue-green + analysis

**Comment by TommyLike** [comment]
@jessesuen fixed in v0.9.2, thanks @jessesuen
